plugins/secrets/update.js

```diff
@@ -0,0 +1,54 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
+'use strict';
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.secret.base, 'id');
+
+module.exports = () => ({
+    method: 'PUT',
+    path: '/secrets/{id}',
+    config: {
+        description: 'Update a secret',
+        notes: 'Update a specific secret',
+        tags: ['api', 'secrets'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        handler: (request, reply) => {
+            const factory = request.server.app.secretFactory;
+            const credentials = request.auth.credentials;
+            const canAccess = request.server.plugins.secrets.canAccess;
+
+            return factory.get(request.params.id)
+                .then(secret => {
+                    if (!secret) {
+                        throw boom.notFound('Secret does not exist');
+                    }
+
+                    Object.keys(request.payload).forEach(key => {
```

Comment:
Should probably move this to after canAccess, right before `secret.update()`. No point of doing this if user doesn't have permission.